jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java

```diff
@@ -887,12 +903,38 @@ public void setDeflaterPoolCapacity(int capacity)
         if (isStarted())
             throw new IllegalStateException(getState());
 
-        poolCapacity = capacity;
+        _deflaterPool.setCapacity(capacity);
+    }
+
+    /**
+     * Gets the maximum number of Inflators that the DeflaterPool can hold.
+     *
+     * @return the Deflater pool capacity
+     */
+    public int getInflaterPoolCapacity()
+    {
+        return _inflaterPool.getCapacity();
+    }
+
+    /**
+     * Sets the maximum number of Inflators that the DeflaterPool can hold.
+     */
+    public void setInflaterPoolCapacity(int capacity)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+
+        _inflaterPool.setCapacity(capacity);
+    }
+
+    protected InflaterPool newInflaterPool()
+    {
+        return new InflaterPool(CompressionPool.INFINITE_CAPACITY, true);
     }
 
-    protected DeflaterPool newDeflaterPool(int capacity)
+    protected DeflaterPool newDeflaterPool()
     {
-        return new DeflaterPool(capacity, Deflater.DEFAULT_COMPRESSION, true);
+        return new DeflaterPool(CompressionPool.INFINITE_CAPACITY, Deflater.DEFAULT_COMPRESSION, true);
```

Comment:
What does "INFINITE_CAPACITY" mean?

Reply:
It means there is no limit on the number of objects allowed to be pooled.
This is now changed later by the setter, as I have made the capacity of the pool configurable after it is constructed. This lets us make the pool final and add it as a bean to the GzipHandler.

Reply:
Opened PR #5295, which uses the new `Pool` class for the `CompressionPool` and still allows the pool size to be changed before it is started.